tests/io/dask/dataframe/test_repartition.py

```diff
@@ -0,0 +1,131 @@
+        partition_on=["part"],
+    )
+
+    repartition_ratio = calculate_repartition_ratio(
+        dataset_factory=dataset_factory_function_scope,
+        table_name=SINGLE_TABLE,
+        dataset_metadata_frac=frac,
+        target_nrows_per_partition=target_nrows_per_partition,
+    ).compute()
+
+    ddf = read_dataset_as_ddf(factory=dataset_factory_function_scope)
+    ddf = repartition_ddf(ddf, repartition_ratio=repartition_ratio)
+
+    assert (ddf.map_partitions(len).compute() == target_nrows_per_partition).all()
+
+
+@pytest.mark.parametrize("frac", (0.3, 0.1))
+@pytest.mark.parametrize(
+    "df,target_nrows_per_partition",
+    (
+        # repartition by decreasing partition size, increasing number of partitions
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 97),
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 31),
+        # repartition by increasing partition size, decreasing number of partitions
+        (pd.DataFrame({"A": range(2 ** 6), "part": list("0123456789abcdef") * 4}), 11),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 7),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 1),
+    ),
+)
+def test_calculate_repartition_ratio_approximate_match(
+    dataset_factory_function_scope, df, target_nrows_per_partition, frac
+):
+    dfs = [df]
+    store_dataframes_as_dataset__iter(
+        dfs,
+        store=dataset_factory_function_scope.store_factory,
+        dataset_uuid=dataset_factory_function_scope.dataset_uuid,
+        partition_on=["part"],
+    )
+
+    repartition_ratio = calculate_repartition_ratio(
+        dataset_factory=dataset_factory_function_scope,
+        table_name=SINGLE_TABLE,
+        dataset_metadata_frac=frac,
+        target_nrows_per_partition=target_nrows_per_partition,
+    ).compute()
+
+    ddf = read_dataset_as_ddf(factory=dataset_factory_function_scope)
+    ddf = repartition_ddf(ddf, repartition_ratio=repartition_ratio)
+    # this is somewhat arbitrary as original partition size may be non-constant
+    margin_of_error = 0.5  # proportion
+    partition_lengths = ddf.map_partitions(len).compute()
+    assert (
+        ((1 - margin_of_error) * target_nrows_per_partition <= partition_lengths)
+        & (partition_lengths <= (1 + margin_of_error) * target_nrows_per_partition)
+    ).all()
+
+
+@pytest.mark.parametrize(
+    "df,repartition_ratio",
+    (  # repartition by decreasing partition size, increasing number of partitions
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 3.88),
+        (
+            pd.DataFrame({"A": range(2 ** 7), "part": ["A", "B", "C", "D"] * 2 ** 5}),
+            0.25,
+        ),
+        # repartition by increasing partition size, decreasing number of partitions
+        (pd.DataFrame({"A": range(50), "part": range(50)}), 7),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 1.0),
```

Comment:
I think we can use a fixed dataset for these tests and test a repartitioning of `<1, ==1, >1`. That would reduce test complexity

Reply:
Done. Please check and confirm.